types.bal

```diff
@@ -155,40 +147,47 @@ public isolated function mapToExperimentDataResponse(database:ExperimentDataFull
     }
     if (inputFor != ()) {
         dataMapped.inputFor = inputFor;
-        dataMapped.inputForLinks = from var step in inputFor select string `/experiments/${data.experimentId}/timeline/${step}`;
+        dataMapped.inputForLinks = from var step in inputFor
+            select string `/experiments/${data.experimentId}/timeline/${step}`;
     }
     return dataMapped;
 }
 
-
 ////////////////////////////////////////////////////////////////////////////////
 // Timeline ////////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////////////
 
 public type TimelineStepPost record {|
     string resultLocation;
     string[] inputData;
-
     string processorName;
-    string? processorVersion=();
-    string? processorLocation=();
+    string? processorVersion = ();
+    string? processorLocation = ();
     string? parameters;
-    string parametersContentType=mime:APPLICATION_FORM_URLENCODED;
-    string? parametersDescriptionLocation=();
-|};
+    string parametersContentType = mime:APPLICATION_FORM_URLENCODED;
+    string? parametersDescriptionLocation = ();
+    int? progressValue = ();
+    int? progressStart = ();
+    int? progressTarget = ();
+    string? progressUnit = ();
+|}; // TODO: not sure if sth needs to be changed here
 
 public type TimelineStepMinResponse record {|
     *ApiResponse;
     string notes;
     int sequence;
     string 'start;
-    string? end=();
+    string? end = ();
     string status;
     string processorName;
-    string? processorVersion=();
-    string? processorLocation=();
-    string? parametersContentType=();
-    string? parametersDescriptionLocation=();
+    string? processorVersion = ();
+    string? processorLocation = ();
+    string? parametersContentType = ();
+    string? parametersDescriptionLocation = ();
```

Comment:
We should deprecate this. I thought it was a good idea, but I don't think it can work with substeps. It was meant to be an URL to the microfrontend so the parameters could be displayed in the micro frontend of the plugin. But with substeps these microfrontends are ephemereal and may disappear. So it is better to have parameter display components for some common content types.

Reply:
Should we remove it or just leave it in for now?

Reply:
we can just remove it. The other changes will break the UI anyway so one more does not hurt.

Reply:
also processorName?

Reply:
No. Processor name and version together uniquely identify the plugin that was used.